homeassistant/components/generic/config_flow.py

```diff
@@ -261,6 +265,17 @@ async def async_test_stream(
     return {}
 
 
+def register_preview(hass: HomeAssistant, flow_id: str, preview: GenericCamera):
+    """Set up previews for camera feeds during config flow."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if not hass.data[DOMAIN].get(PREVIEWS):
+        _LOGGER.debug("Registering camera image preview handler")
+        hass.http.register_view(CameraImagePreview(hass))
+        hass.data[DOMAIN][PREVIEWS] = {}
+    hass.data[DOMAIN][PREVIEWS][flow_id] = preview
```

Comment:
I'm thinking this ideally should get cleaned up as well at some point, though i'm not sure it's worth the code to do it if we keep this tiny or bounded.

One other thought: did you consider storing the "user_input" here instead of the camera object? Then create the camera object in the preview based on the config. 

I could also see us changing the config flow api in the future to have a dictionary of data about the current state that could be exposed via one of the real config flow APIs (could avoid need for cleanup) and so just storing the dict data could allow for something like that.

Reply:
Agreed, done.